scripts/composer/ScriptHandler.php

```diff
@@ -88,6 +88,77 @@ public static function downloadDevelopPackages(Event $event) {
     }
   }
 
+  /**
+   * Reset local repositories to the default branch.
+   *
+   * @param \Composer\Script\Event $event
+   *   The script event.
+   */
+  public static function resetLocalRepositories(Event $event) {
+    $io = $event->getIO();
+    $repositoriesInfo = self::getLocalRepositoriesInfo($event);
+    $drupalFinder = new DrupalFinder();
+    $drupalFinder->locateRoot(getcwd());
+    $composerRoot = $drupalFinder->getComposerRoot();
+
+    $io->write(PHP_EOL);
+    foreach ($repositoriesInfo as $key => $info) {
+      $localBranch = trim(shell_exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' rev-parse --abbrev-ref HEAD'));
+
+      if ($localBranch === $info['branch']) {
```

Comment:
If I'm already on same branch, but there are some changes in it -> I would expect that branch is reseted, but this `if` will ignore it.

I think it should be combined with `$gitStatus` and only if branch is same and there are no changes, then we can `continue`.

Reply:
I am not sure about this, I was thinking, that you might be working in that project, and just want to reset all other projects, be up to date.